indexer/data_header.hpp

```diff
@@ -83,9 +85,13 @@ namespace feature
     inline void SetType(MapType t) { m_type = t; }
     inline MapType GetType() const { return m_type; }
 
+    inline RegionData const & GetData() const { return m_data; }
+    inline void SetData(RegionData & data) { swap(m_data, data); }
```

Comment:
No, please. Use rvalue-reference or copy here. Swap may lead to hard-to-catch errors.